aws-codeartifact-repository/src/main/java/software/amazon/codeartifact/repository/UpdateHandler.java

```diff
@@ -22,88 +32,145 @@
+            .makeServiceCall((awsRequest, client) -> {
+                AwsResponse awsResponse = null;
+                try {
+                    awsResponse = client.injectCredentialsAndInvokeV2(awsRequest, client.client()::updateRepository);
+                } catch (final AwsServiceException e) {
+                    String repositoryName = progress.getResourceModel().getRepositoryName();
+                    Translator.throwCfnException(e, Constants.CREATE_REPOSITORY, repositoryName);
+                }
+                logger.log(String.format("%s successfully created.", ResourceModel.TYPE_NAME));
+                return awsResponse;
+            })
+            .progress();
+    }
+
+    protected ProgressEvent<ResourceModel, CallbackContext> updateRepositoryPermissionsPolicy(
+        final AmazonWebServicesClientProxy proxy,
+        final ProgressEvent<ResourceModel, CallbackContext> progress,
+        final CallbackContext callbackContext,
+        final ResourceHandlerRequest<ResourceModel> request,
+        final ProxyClient<CodeartifactClient> proxyClient,
+        final Logger logger
+    ) {
+        final ResourceModel desiredModel = progress.getResourceModel();
+        if (desiredModel.getPolicyDocument() != null) {
+            return putRepositoryPermissionsPolicy(proxy, progress, callbackContext, proxyClient, logger);
+        }
+        return deleteRepositoryPermissionsPolicy(proxy, progress, callbackContext, proxyClient, logger);
+    }
+
+    protected ProgressEvent<ResourceModel, CallbackContext> deleteRepositoryPermissionsPolicy(
+        final AmazonWebServicesClientProxy proxy,
+        final ProgressEvent<ResourceModel, CallbackContext> progress,
+        final CallbackContext callbackContext,
+        final ProxyClient<CodeartifactClient> proxyClient,
+        final Logger logger
+    ) {
+        final ResourceModel desiredModel = progress.getResourceModel();
+
+        if (desiredModel.getPolicyDocument() != null) {
+            return ProgressEvent.progress(desiredModel, callbackContext);
+        }
+
+        return proxy.initiate("AWS-CodeArtifact-Repository::Update::DeleteRepositoryPermissionsPolicy", proxyClient,
+            progress.getResourceModel(), progress.getCallbackContext())
+            .translateToServiceRequest(Translator::translateDeletePermissionsPolicyRequest)
+            .makeServiceCall((awsRequest, client) -> {
+                DeleteRepositoryPermissionsPolicyResponse awsResponse = null;
+                try {
+                    awsResponse = client.injectCredentialsAndInvokeV2(awsRequest, client.client()::deleteRepositoryPermissionsPolicy);
+                    logger.log("Repository permission policy successfully deleted.");
+                } catch (final AwsServiceException e) {
+                    String domainName = desiredModel.getDomainName();
+                    Translator.throwCfnException(e, Constants.PUT_REPOSITORY_POLICY, domainName);
```

Comment:
Technically not put repo policy, but delete

Reply:
thanks, will update